src/main/java/com/icthh/xm/uaa/service/query/UserQueryService.java

```diff
@@ -169,4 +181,53 @@ protected <X> Specification<User> valueIn(Function<Root<User>, Expression<X>> me
             return predicate;
         };
     }
+
+    private List<Optional<Specification<User>>> buildDataAttributes(List<DataAttributeCriteria> dataAttributes) {
+        List<Optional<Specification<User>>> specs = new ArrayList<>();
+        for (DataAttributeCriteria dataAttributeCriteria : dataAttributes) {
+            Specification<User> spec = buildDataSpecification(dataAttributeCriteria);
+            specs.add(Optional.of(spec));
+        }
+        return specs;
+    }
```

Comment:
Method can be simplified as: 
```
    private Stream<Optional<Specification<User>>> buildDataAttributes(List<DataAttributeCriteria> dataAttributes) {
        return dataAttributes.stream()
                             .map(this::buildDataSpecification)
                             .map(Optional::ofNullable);
    }
    ```
    
    1. return type is already stream - do not need to convert on calling side
    2. Functional style without external collection mutation and for loop.